src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -0,0 +1,366 @@
+
+            using (var getObjectIdCommand = new SqlCommand(getObjectIdQuery, connection))
+            using (SqlDataReader reader = await getObjectIdCommand.ExecuteReaderAsync(cancellationToken))
+            {
+                if (!await reader.ReadAsync(cancellationToken))
+                {
+                    throw new InvalidOperationException($"Received empty response when querying the object ID for table: '{this._userTable.FullName}'.");
+                }
+
+                object userTableId = reader.GetValue(0);
+
+                if (userTableId is DBNull)
+                {
+                    throw new InvalidOperationException($"Could not find table: '{this._userTable.FullName}'.");
+                }
+
+                return (int)userTableId;
+            }
+        }
+
+        /// <summary>
+        /// Gets the names and types of primary key columns of the user table.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if there are no primary key columns present in the user table or if their names conflict with columns in worker table.
+        /// </exception>
+        private async Task<IReadOnlyList<(string name, string type)>> GetPrimaryKeyColumnsAsync(SqlConnection connection, int userTableId, CancellationToken cancellationToken)
+        {
+            string getPrimaryKeyColumnsQuery = $@"
+                SELECT c.name, t.name, c.max_length, c.precision, c.scale
+                FROM sys.indexes AS i
+                INNER JOIN sys.index_columns AS ic ON i.object_id = ic.object_id AND i.index_id = ic.index_id
+                INNER JOIN sys.columns AS c ON ic.object_id = c.object_id AND ic.column_id = c.column_id
+                INNER JOIN sys.types AS t ON c.user_type_id = t.user_type_id
+                WHERE i.is_primary_key = 1 AND i.object_id = {userTableId};
+            ";
+
+            using (var getPrimaryKeyColumnsCommand = new SqlCommand(getPrimaryKeyColumnsQuery, connection))
+            using (SqlDataReader reader = await getPrimaryKeyColumnsCommand.ExecuteReaderAsync(cancellationToken))
+            {
+                string[] reservedColumnNames = new string[] { "ChangeVersion", "AttemptCount", "LeaseExpirationTime" };
+                string[] variableLengthTypes = new string[] { "varchar", "nvarchar", "nchar", "char", "binary", "varbinary" };
+                string[] variablePrecisionTypes = new string[] { "numeric", "decimal" };
+
+                var primaryKeyColumns = new List<(string name, string type)>();
+
+                while (await reader.ReadAsync(cancellationToken))
+                {
+                    string name = reader.GetString(0);
+
+                    if (reservedColumnNames.Contains(name))
+                    {
+                        throw new InvalidOperationException($"Found reserved column name: '{name}' in table: '{this._userTable.FullName}'.");
```

Comment:
1. Let's make sure this is documented in the README
2. I'd suggest adding a bit more text to tell the user how to resolve it, I think it's a bit unclear right now what the user can do about it. Something like `This column must be removed or renamed for the table to be used by this trigger.`
3. I'd also make it find all of the columns that are reserved and list them, not just the first one

Reply:
This is resolved in new PR #288